components/badge/badge_constants.js

```diff
@@ -12,3 +12,29 @@ export const BADGE_KIND_MODIFIERS = {
   label: '',
   count: 'd-badge--count',
 };
+
+export const BADGE_DECORATION_MODIFIERS = {
+  default: '',
```

Comment:
Make this null or undefined rather than empty string or else it will actually insert an empty string into the class list, which isn't that big of deal but it's still a bit messy.

Reply:
mhmm I cannot reproduce the blank space, it is not being added if its default

<img width="1155" alt="image" src="https://user-images.githubusercontent.com/89984179/225707612-de3ddeee-6b63-4333-812d-389b878d60cd.png">

Reply:
Removed this from here. Made default to be undefined.